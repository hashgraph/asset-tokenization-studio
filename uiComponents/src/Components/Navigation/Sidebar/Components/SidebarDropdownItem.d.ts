/// <reference types="react" />
import type { PhosphorIconProps } from "@/Components/Foundations/PhosphorIcon";
import type { MenuItemProps as ChakraMenuItemProps } from "@chakra-ui/react";
export interface SidebarDropdownItemProps extends ChakraMenuItemProps {
    label: string;
    icon?: PhosphorIconProps["as"];
    isDisabled?: boolean;
    isActive?: boolean;
}
export declare const SidebarDropdownItem: ({ label, icon, isDisabled, isActive, ...props }: SidebarDropdownItemProps) => JSX.Element;
