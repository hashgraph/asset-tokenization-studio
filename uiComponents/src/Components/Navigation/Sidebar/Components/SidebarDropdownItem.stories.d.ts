declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("./SidebarDropdownItem").SidebarDropdownItemProps>;
export default meta;
export declare const Default: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./SidebarDropdownItem").SidebarDropdownItemProps>;
export declare const WithoutIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("./SidebarDropdownItem").SidebarDropdownItemProps>;
