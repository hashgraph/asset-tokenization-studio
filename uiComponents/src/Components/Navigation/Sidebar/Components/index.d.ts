export * from "./SidebarDropdownItem";
export * from "./SidebarItem";
export * from "./SidebarItemWithDropdown";
