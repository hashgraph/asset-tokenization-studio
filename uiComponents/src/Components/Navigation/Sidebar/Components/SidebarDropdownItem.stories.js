var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { SidebarDropdownItem } from "./SidebarDropdownItem";
import { Cookie } from "@phosphor-icons/react";
import { Menu } from "@chakra-ui/react";
var meta = {
    title: "Design System/Navigation/Sidebar/SidebarDropdownItem",
    component: SidebarDropdownItem,
    argTypes: {
        isActive: {
            control: "boolean",
            description: "Boolean to specify if the item is active."
        },
        isDisabled: {
            control: "boolean",
            description: "Boolean to specify if the item is disabled."
        },
        icon: { control: false, description: "Icon to show in the item" }
    },
    parameters: {
        docs: {}
    },
    args: {
        label: "Item molecule",
        icon: Cookie
    }
};
export default meta;
var Template = function (args) {
    return (React.createElement(Menu, null,
        React.createElement(SidebarDropdownItem, __assign({}, args))));
};
export var Default = Template.bind({});
Default.args = {};
export var WithoutIcon = Template.bind({});
WithoutIcon.args = {
    icon: null
};
