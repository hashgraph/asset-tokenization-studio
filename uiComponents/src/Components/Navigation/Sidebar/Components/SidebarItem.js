var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
import { forwardRef, useMultiStyleConfig as useChakraMultiStyleConfig, } from "@chakra-ui/system";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Text } from "@/Components/Foundations/Text";
import { useHover } from "@/Hooks";
import { Center as ChakraCenter, Flex as ChakraFlex, useMergeRefs as useChakraMergeRefs, } from "@chakra-ui/react";
export var sidebarPartsList = ["iconWeight", "icon", "label", "container"];
export var SidebarItem = forwardRef(function (_a, ref) {
    var isActive = _a.isActive, isDisabled = _a.isDisabled, label = _a.label, icon = _a.icon, props = __rest(_a, ["isActive", "isDisabled", "label", "icon"]);
    var _b = useHover(), containerRef = _b[0], isHovered = _b[1];
    var refs = useChakraMergeRefs(containerRef, ref);
    var styles = useChakraMultiStyleConfig("SidebarItem", {
        isHovered: isHovered,
        isActive: isActive,
        isDisabled: isDisabled
    });
    return (React.createElement(ChakraFlex, __assign({ align: "center", direction: "column", outline: "none", "data-testid": "sidebar-item-".concat(label), disabled: isDisabled, ref: refs }, props, { sx: styles.container }),
        React.createElement(ChakraCenter, { as: "button", sx: styles.icon },
            React.createElement(PhosphorIcon, { size: "xs", as: icon, weight: styles.iconWeight })),
        React.createElement(Text, { sx: styles.label },
            label,
            " ")));
});
