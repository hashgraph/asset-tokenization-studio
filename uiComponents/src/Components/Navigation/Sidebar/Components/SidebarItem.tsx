import React from "react";
import {
  forwardRef,
  useMultiStyleConfig as useChakraMultiStyleConfig,
} from "@chakra-ui/system";
import type {
  PhosphorIconProps,
  Weight,
} from "@/Components/Foundations/PhosphorIcon";
import { PhosphorIcon } from "@/Components/Foundations/PhosphorIcon";
import { Text } from "@/Components/Foundations/Text";
import { useHover } from "@/Hooks";
import type { FlexProps as ChakraFlexProps } from "@chakra-ui/react";
import {
  Center as ChakraCenter,
  Flex as ChakraFlex,
  useMergeRefs as useChakraMergeRefs,
} from "@chakra-ui/react";
import type {
  SystemStyleObject as ChakraSystemStyleObject,
  ComponentWithAs,
} from "@chakra-ui/system";
import type {
  BaseMultiStyleConfiguration,
  PartsStyleInterpolation,
} from "@/Theme/Components/BaseMultiStyleConfiguration";
export const sidebarPartsList: Array<
  "iconWeight" | "icon" | "label" | "container"
> = ["iconWeight", "icon", "label", "container"];

type Parts = typeof sidebarPartsList;

export type SidebarItemConfigProps = {
  isDisabled?: boolean;
  isHovered?: boolean;
  isActive?: boolean;
};

export interface SidebarItemThemeConfiguration
  extends BaseMultiStyleConfiguration<Parts> {
  baseStyle:
    | (({
        isHovered,
        isDisabled,
        isActive,
      }: SidebarItemConfigProps) => Partial<SidebarItemThemeStyle>)
    | PartsStyleInterpolation<Parts>;
}

export interface SidebarItemProps extends ChakraFlexProps {
  isActive?: boolean;
  isDisabled?: boolean;
  label: string;
  icon: PhosphorIconProps["as"];
}

export type SidebarItemThemeStyle = Record<
  Parts[number],
  ChakraSystemStyleObject
> & {
  iconWeight: Weight;
};

export const SidebarItem: ComponentWithAs<"button", SidebarItemProps> =
  forwardRef<SidebarItemProps, "button">(
    (
      { isActive, isDisabled, label, icon, ...props }: SidebarItemProps,
      ref
    ) => {
      const [containerRef, isHovered] = useHover();
      const refs = useChakraMergeRefs(containerRef, ref);
      const styles = useChakraMultiStyleConfig("SidebarItem", {
        isHovered,
        isActive,
        isDisabled,
      }) as SidebarItemThemeStyle;

      return (
        <ChakraFlex
          align="center"
          direction="column"
          outline="none"
          // as="button"
          data-testid={`sidebar-item-${label}`}
          disabled={isDisabled}
          ref={refs}
          {...props}
          sx={styles.container}
        >
          <ChakraCenter as="button" sx={styles.icon}>
            <PhosphorIcon size="xs" as={icon} weight={styles.iconWeight} />
          </ChakraCenter>
          <Text sx={styles.label}>{label} </Text>
        </ChakraFlex>
      );
    }
  );
