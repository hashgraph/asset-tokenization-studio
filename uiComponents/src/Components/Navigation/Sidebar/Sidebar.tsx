import React from "react";
import { useStyleConfig as useChakraStyleConfig } from "@chakra-ui/system";
import type { FlexProps } from "@chakra-ui/react";
import { Flex as ChakraFlex, Spacer as ChakraSpacer } from "@chakra-ui/react";

export interface SidebarProps extends FlexProps {
  topContent?: React.ReactElement;
  bottomContent?: React.ReactElement;
}
export const Sidebar = ({
  topContent,
  bottomContent,
  ...props
}: SidebarProps) => {
  const styles = useChakraStyleConfig("Sidebar");

  return (
    <ChakraFlex sx={styles} {...props}>
      {topContent || <ChakraSpacer />}
      {bottomContent}
    </ChakraFlex>
  );
};
