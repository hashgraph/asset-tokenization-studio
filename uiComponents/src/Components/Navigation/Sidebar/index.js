export * from "./Components";
export * from "./Sidebar";
