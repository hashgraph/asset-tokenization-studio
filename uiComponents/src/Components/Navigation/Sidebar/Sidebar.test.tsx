import { render } from "@/test-utils";
import { Cookie, Gear, House } from "@phosphor-icons/react";
import React from "react";
import type { SidebarProps } from "./Sidebar";
import { Sidebar } from "./Sidebar";
import { Stack } from "@chakra-ui/react";
import { SidebarItem } from "./Components/SidebarItem";

describe(`<Sidebar />`, () => {
  const defaultProps: SidebarProps = {
    topContent: (
      <Stack spacing={6}>
        <SidebarItem label="Home" icon={House} />
        <SidebarItem label="Item molecule" icon={Cookie} isActive />
        <SidebarItem label="Item molecule" icon={Cookie} isDisabled />
      </Stack>
    ),
    bottomContent: <SidebarItem label="Configuration" icon={Gear} />,
  };
  const factoryComponent = (props: SidebarProps = defaultProps) =>
    render(<Sidebar {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });
  test("renders correctly only top content", () => {
    const component = factoryComponent({ topContent: defaultProps.topContent });

    expect(component.asFragment()).toMatchSnapshot("OnlyTopContent");
  });

  test("renders correctly only bottom content", () => {
    const component = factoryComponent({
      topContent: defaultProps.bottomContent,
    });

    expect(component.asFragment()).toMatchSnapshot("OnlyBottomContent");
  });
});
