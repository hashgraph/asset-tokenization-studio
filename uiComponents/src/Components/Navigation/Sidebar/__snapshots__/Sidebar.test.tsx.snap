// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1s7szjn"
  >
    <div
      class="chakra-stack css-b5leqj"
    >
      <div
        class="css-df6r3v"
        data-testid="sidebar-item-Home"
      >
        <button
          class="css-1rb62th"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M218.83,103.77l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208a16,16,0,0,0,16,16H96a16,16,0,0,0,16-16V160h32v48a16,16,0,0,0,16,16h48a16,16,0,0,0,16-16V115.55A16,16,0,0,0,218.83,103.77ZM208,208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48V115.55l.11-.1L128,40l79.9,75.43.11.1Z"
            />
          </svg>
        </button>
        <p
          class="css-owi0ee"
        >
          Home 
        </p>
      </div>
      <div
        class="css-df6r3v"
        data-testid="sidebar-item-Item molecule"
      >
        <button
          class="css-51z7wg"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224,120a40,40,0,0,1-40-40,8,8,0,0,0-8-8,40,40,0,0,1-40-40,8,8,0,0,0-8-8A104,104,0,1,0,232,128,8,8,0,0,0,224,120ZM75.51,99.51a12,12,0,1,1,0,17A12,12,0,0,1,75.51,99.51Zm25,73a12,12,0,1,1,0-17A12,12,0,0,1,100.49,172.49Zm23-40a12,12,0,1,1,17,0A12,12,0,0,1,123.51,132.49Zm41,48a12,12,0,1,1,0-17A12,12,0,0,1,164.49,180.49Z"
            />
          </svg>
        </button>
        <p
          class="css-atzlkw"
        >
          Item molecule 
        </p>
      </div>
      <div
        class="css-1m0okgb"
        data-testid="sidebar-item-Item molecule"
        disabled=""
      >
        <button
          class="css-u61f6f"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
            />
          </svg>
        </button>
        <p
          class="css-t8zpnn"
        >
          Item molecule 
        </p>
      </div>
    </div>
    <div
      class="css-df6r3v"
      data-testid="sidebar-item-Configuration"
    >
      <button
        class="css-1rb62th"
      >
        <svg
          class="chakra-icon css-1qzovvq"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128,80a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Zm88-29.84q.06-2.16,0-4.32l14.92-18.64a8,8,0,0,0,1.48-7.06,107.21,107.21,0,0,0-10.88-26.25,8,8,0,0,0-6-3.93l-23.72-2.64q-1.48-1.56-3-3L186,40.54a8,8,0,0,0-3.94-6,107.71,107.71,0,0,0-26.25-10.87,8,8,0,0,0-7.06,1.49L130.16,40Q128,40,125.84,40L107.2,25.11a8,8,0,0,0-7.06-1.48A107.6,107.6,0,0,0,73.89,34.51a8,8,0,0,0-3.93,6L67.32,64.27q-1.56,1.49-3,3L40.54,70a8,8,0,0,0-6,3.94,107.71,107.71,0,0,0-10.87,26.25,8,8,0,0,0,1.49,7.06L40,125.84Q40,128,40,130.16L25.11,148.8a8,8,0,0,0-1.48,7.06,107.21,107.21,0,0,0,10.88,26.25,8,8,0,0,0,6,3.93l23.72,2.64q1.49,1.56,3,3L70,215.46a8,8,0,0,0,3.94,6,107.71,107.71,0,0,0,26.25,10.87,8,8,0,0,0,7.06-1.49L125.84,216q2.16.06,4.32,0l18.64,14.92a8,8,0,0,0,7.06,1.48,107.21,107.21,0,0,0,26.25-10.88,8,8,0,0,0,3.93-6l2.64-23.72q1.56-1.48,3-3L215.46,186a8,8,0,0,0,6-3.94,107.71,107.71,0,0,0,10.87-26.25,8,8,0,0,0-1.49-7.06Zm-16.1-6.5a73.93,73.93,0,0,1,0,8.68,8,8,0,0,0,1.74,5.48l14.19,17.73a91.57,91.57,0,0,1-6.23,15L187,173.11a8,8,0,0,0-5.1,2.64,74.11,74.11,0,0,1-6.14,6.14,8,8,0,0,0-2.64,5.1l-2.51,22.58a91.32,91.32,0,0,1-15,6.23l-17.74-14.19a8,8,0,0,0-5-1.75h-.48a73.93,73.93,0,0,1-8.68,0,8,8,0,0,0-5.48,1.74L100.45,215.8a91.57,91.57,0,0,1-15-6.23L82.89,187a8,8,0,0,0-2.64-5.1,74.11,74.11,0,0,1-6.14-6.14,8,8,0,0,0-5.1-2.64L46.43,170.6a91.32,91.32,0,0,1-6.23-15l14.19-17.74a8,8,0,0,0,1.74-5.48,73.93,73.93,0,0,1,0-8.68,8,8,0,0,0-1.74-5.48L40.2,100.45a91.57,91.57,0,0,1,6.23-15L69,82.89a8,8,0,0,0,5.1-2.64,74.11,74.11,0,0,1,6.14-6.14A8,8,0,0,0,82.89,69L85.4,46.43a91.32,91.32,0,0,1,15-6.23l17.74,14.19a8,8,0,0,0,5.48,1.74,73.93,73.93,0,0,1,8.68,0,8,8,0,0,0,5.48-1.74L155.55,40.2a91.57,91.57,0,0,1,15,6.23L173.11,69a8,8,0,0,0,2.64,5.1,74.11,74.11,0,0,1,6.14,6.14,8,8,0,0,0,5.1,2.64l22.58,2.51a91.32,91.32,0,0,1,6.23,15l-14.19,17.74A8,8,0,0,0,199.87,123.66Z"
          />
        </svg>
      </button>
      <p
        class="css-owi0ee"
      >
        Configuration 
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Sidebar /> renders correctly only bottom content: OnlyBottomContent 1`] = `
<DocumentFragment>
  <div
    class="css-1s7szjn"
  >
    <div
      class="css-df6r3v"
      data-testid="sidebar-item-Configuration"
    >
      <button
        class="css-1rb62th"
      >
        <svg
          class="chakra-icon css-1qzovvq"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="0 0 256 256"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M128,80a48,48,0,1,0,48,48A48.05,48.05,0,0,0,128,80Zm0,80a32,32,0,1,1,32-32A32,32,0,0,1,128,160Zm88-29.84q.06-2.16,0-4.32l14.92-18.64a8,8,0,0,0,1.48-7.06,107.21,107.21,0,0,0-10.88-26.25,8,8,0,0,0-6-3.93l-23.72-2.64q-1.48-1.56-3-3L186,40.54a8,8,0,0,0-3.94-6,107.71,107.71,0,0,0-26.25-10.87,8,8,0,0,0-7.06,1.49L130.16,40Q128,40,125.84,40L107.2,25.11a8,8,0,0,0-7.06-1.48A107.6,107.6,0,0,0,73.89,34.51a8,8,0,0,0-3.93,6L67.32,64.27q-1.56,1.49-3,3L40.54,70a8,8,0,0,0-6,3.94,107.71,107.71,0,0,0-10.87,26.25,8,8,0,0,0,1.49,7.06L40,125.84Q40,128,40,130.16L25.11,148.8a8,8,0,0,0-1.48,7.06,107.21,107.21,0,0,0,10.88,26.25,8,8,0,0,0,6,3.93l23.72,2.64q1.49,1.56,3,3L70,215.46a8,8,0,0,0,3.94,6,107.71,107.71,0,0,0,26.25,10.87,8,8,0,0,0,7.06-1.49L125.84,216q2.16.06,4.32,0l18.64,14.92a8,8,0,0,0,7.06,1.48,107.21,107.21,0,0,0,26.25-10.88,8,8,0,0,0,3.93-6l2.64-23.72q1.56-1.48,3-3L215.46,186a8,8,0,0,0,6-3.94,107.71,107.71,0,0,0,10.87-26.25,8,8,0,0,0-1.49-7.06Zm-16.1-6.5a73.93,73.93,0,0,1,0,8.68,8,8,0,0,0,1.74,5.48l14.19,17.73a91.57,91.57,0,0,1-6.23,15L187,173.11a8,8,0,0,0-5.1,2.64,74.11,74.11,0,0,1-6.14,6.14,8,8,0,0,0-2.64,5.1l-2.51,22.58a91.32,91.32,0,0,1-15,6.23l-17.74-14.19a8,8,0,0,0-5-1.75h-.48a73.93,73.93,0,0,1-8.68,0,8,8,0,0,0-5.48,1.74L100.45,215.8a91.57,91.57,0,0,1-15-6.23L82.89,187a8,8,0,0,0-2.64-5.1,74.11,74.11,0,0,1-6.14-6.14,8,8,0,0,0-5.1-2.64L46.43,170.6a91.32,91.32,0,0,1-6.23-15l14.19-17.74a8,8,0,0,0,1.74-5.48,73.93,73.93,0,0,1,0-8.68,8,8,0,0,0-1.74-5.48L40.2,100.45a91.57,91.57,0,0,1,6.23-15L69,82.89a8,8,0,0,0,5.1-2.64,74.11,74.11,0,0,1,6.14-6.14A8,8,0,0,0,82.89,69L85.4,46.43a91.32,91.32,0,0,1,15-6.23l17.74,14.19a8,8,0,0,0,5.48,1.74,73.93,73.93,0,0,1,8.68,0,8,8,0,0,0,5.48-1.74L155.55,40.2a91.57,91.57,0,0,1,15,6.23L173.11,69a8,8,0,0,0,2.64,5.1,74.11,74.11,0,0,1,6.14,6.14,8,8,0,0,0,5.1,2.64l22.58,2.51a91.32,91.32,0,0,1,6.23,15l-14.19,17.74A8,8,0,0,0,199.87,123.66Z"
          />
        </svg>
      </button>
      <p
        class="css-owi0ee"
      >
        Configuration 
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Sidebar /> renders correctly only top content: OnlyTopContent 1`] = `
<DocumentFragment>
  <div
    class="css-1s7szjn"
  >
    <div
      class="chakra-stack css-b5leqj"
    >
      <div
        class="css-df6r3v"
        data-testid="sidebar-item-Home"
      >
        <button
          class="css-1rb62th"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M218.83,103.77l-80-75.48a1.14,1.14,0,0,1-.11-.11,16,16,0,0,0-21.53,0l-.11.11L37.17,103.77A16,16,0,0,0,32,115.55V208a16,16,0,0,0,16,16H96a16,16,0,0,0,16-16V160h32v48a16,16,0,0,0,16,16h48a16,16,0,0,0,16-16V115.55A16,16,0,0,0,218.83,103.77ZM208,208H160V160a16,16,0,0,0-16-16H112a16,16,0,0,0-16,16v48H48V115.55l.11-.1L128,40l79.9,75.43.11.1Z"
            />
          </svg>
        </button>
        <p
          class="css-owi0ee"
        >
          Home 
        </p>
      </div>
      <div
        class="css-df6r3v"
        data-testid="sidebar-item-Item molecule"
      >
        <button
          class="css-51z7wg"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224,120a40,40,0,0,1-40-40,8,8,0,0,0-8-8,40,40,0,0,1-40-40,8,8,0,0,0-8-8A104,104,0,1,0,232,128,8,8,0,0,0,224,120ZM75.51,99.51a12,12,0,1,1,0,17A12,12,0,0,1,75.51,99.51Zm25,73a12,12,0,1,1,0-17A12,12,0,0,1,100.49,172.49Zm23-40a12,12,0,1,1,17,0A12,12,0,0,1,123.51,132.49Zm41,48a12,12,0,1,1,0-17A12,12,0,0,1,164.49,180.49Z"
            />
          </svg>
        </button>
        <p
          class="css-atzlkw"
        >
          Item molecule 
        </p>
      </div>
      <div
        class="css-1m0okgb"
        data-testid="sidebar-item-Item molecule"
        disabled=""
      >
        <button
          class="css-u61f6f"
        >
          <svg
            class="chakra-icon css-1qzovvq"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M164.49,163.51a12,12,0,1,1-17,0A12,12,0,0,1,164.49,163.51Zm-81-8a12,12,0,1,0,17,0A12,12,0,0,0,83.51,155.51Zm9-39a12,12,0,1,0-17,0A12,12,0,0,0,92.49,116.49Zm48-1a12,12,0,1,0,0,17A12,12,0,0,0,140.49,115.51ZM232,128A104,104,0,1,1,128,24a8,8,0,0,1,8,8,40,40,0,0,0,40,40,8,8,0,0,1,8,8,40,40,0,0,0,40,40A8,8,0,0,1,232,128Zm-16.31,7.39A56.13,56.13,0,0,1,168.5,87.5a56.13,56.13,0,0,1-47.89-47.19,88,88,0,1,0,95.08,95.08Z"
            />
          </svg>
        </button>
        <p
          class="css-t8zpnn"
        >
          Item molecule 
        </p>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
